appeals/web/src/server/appeals/appeal-details/service-user/__tests__/service-user.test.js

```diff
@@ -0,0 +1,212 @@
+import { parseHtml } from '@pins/platform';
+import supertest from 'supertest';
+import { appealData } from '#testing/app/fixtures/referencedata.js';
+import { createTestEnvironment } from '#testing/index.js';
+import nock from 'nock';
+
+const { app, installMockApi, teardown } = createTestEnvironment();
+const request = supertest(app);
+const baseUrl = '/appeals-service/appeal-details';
+
+describe('service-user', () => {
+	beforeEach(installMockApi);
+	afterEach(teardown);
+
+	describe('GET /change/:userType', () => {
+		it('should render the change service user page for an agent', async () => {
+			const appealId = appealData.appealId.toString();
+			const response = await request.get(`${baseUrl}/${appealId}/service-user/change/agent`);
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+			expect(element.innerHTML).toContain(`Update agent details`);
+		});
+
+		it('should render the change service user page for an appellant', async () => {
+			const appealId = appealData.appealId.toString();
+			const response = await request.get(`${baseUrl}/${appealId}/service-user/change/appellant`);
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+			expect(element.innerHTML).toContain(`Update appellant details`);
+		});
+
+		it('should fail to render the change page when the service user type is not a valid string', async () => {
+			const appealId = appealData.appealId.toString();
+			const response = await request.get(`${baseUrl}/${appealId}/service-user/change/fail`);
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+			expect(element.innerHTML).not.toContain(`Update appellant details`);
+			expect(element.innerHTML).not.toContain(`Update agent details`);
+		});
+	});
+
+	describe('POST /change/:userType', () => {
+		it('should re-render changeServiceUser if firstName is null', async () => {
```

Comment:
Very minor, but I generally like to include a reference to there being an error message displayed for these tests - eg. `should re-render changeServiceUser with the expected error if firstName is null`